### ISSUE TITLE
Validation in create group API

### ISSUE THREAD

**Issue by ClawedCatalyst** [Bulkmailer/BulkMailer-Backend#9]
In `mailer/views.py`, at line 31, there is a POST API for creating a group.
Currently, there seems to be no validation regarding the user ID.

`Here's the issue:`

I can simply pass a user token and create a group with a different user ID. Consequently, the group is added to the account associated with the user ID that was passed.

`Here's what should happen:`

When a different user's ID is passed, it should raise an error, indicating 'permission denied.'"

<img width="750" alt="image" src="https://github.com/Bulkmailer/BulkMailer-Backend/assets/97229491/45c5033d-3ce9-4ae2-ba01-e708f4d5298c">

**Comment by lakshay-saini-au8** [comment]
@ClawedCatalyst  I would like work on this

**Comment by ClawedCatalyst** [comment]
[lakshay-saini-au8](https://github.com/lakshay-saini-au8) sure you can

**Comment by kiranrokkam09** [comment]
I would like to contribute to this issue. Is anybody working with it?

**Comment by ClawedCatalyst** [comment]
you can start on this @kiranrokkam09 :)

**Comment by kiranrokkam09** [comment]
You are saying that only authenticated users can make a POST request to the endpoint.

**Comment by ClawedCatalyst** [comment]
Yes only authenticated and id of user should be the authenticated user.
If I pass a different user id it should raise an error @kiranrokkam09

**Comment by kiranrokkam09** [comment]
@ClawedCatalyst you can check the pull request made by me. I added the following code:
![views py - Visual Studio Code 05-10-2023 12_47_20 PM](https://github.com/Bulkmailer/BulkMailer-Backend/assets/85286397/a6b54bf4-3481-4c2c-9691-c18a313aebe5)

**Comment by lakshay-saini-au8** [comment]
@ClawedCatalyst Sorry for the dealy 
 here is the change please do review https://github.com/Bulkmailer/BulkMailer-Backend/pull/17
